native/src/main/java/io/ballerina/stdlib/kafka/utils/KafkaUtils.java

```diff
@@ -624,28 +624,47 @@ private static void handleSupportedTypesForHeaders(String key, ArrayList<byte[]>
         if (appropriateType instanceof ArrayType arrayType) {
             handleHeaderValuesWithArrayType(key, list, arrayType, bHeaderMap);
         } else if (appropriateType.getTag() == STRING_TAG) {
-            bHeaderMap.put(StringUtils.fromString(key), StringUtils.fromString(new String(list.get(0),
+            byte[] value = list.get(0);
```

Comment:
Do we need to validate for length > 0 before accessing the 0th element?

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerinax-kafka/pull/1258/commits/5fb9c7bdf08db5df4e97d770c801d11f16577e52